lib/generate_board.rb

```diff
@@ -1,29 +1,50 @@
 class GenerateBoard
   attr_reader :board, :blank_board, :number_of_mines
-  
-  def initialize(width:, height:, number_of_mines:)
-    @width, @height, @number_of_mines = width, height, number_of_mines
-    @mine_count = 0
-    @board = Array.new(@height) { Array.new(@width) { 0 } }
-    @blank_board = @board.dup
+
+  LIMIT = 5000.0
+
+  def initialize(width:, height:, number_of_mines:, board:, amount: 1)
+    @width, @height, @number_of_mines, @board, @amount = width, height, number_of_mines, board, amount
+    calculate_jobs
     set_board
   end
 
   private
 
-  def set_board 
-    mine_count = 0
-
-    while mine_count != @number_of_mines do
-      # Random Height
-      height = rand(0...@height)
-      # Random Width
-      width = rand(0...@width)
-      # Check Uniqueness
-      unless board[height][width] == 1
-        board[height][width] = 1
-        mine_count += 1
-      end
+  def set_board
+   if @number_of_mines < LIMIT
+     single_record_insertions
+   else
+     multiple_record_insertions
+   end 
+  end
+
+  def multiple_record_insertions
+    @x_jobs.times do
+      make_mine_job(@x)
     end
+
+    @x_1_jobs.times do
+      make_mine_job(@x + 1)
+    end
+  end
+
+  def single_record_insertions
+    @number_of_mines.times do
+      make_mine_job(@amount)
+    end
+  end
+
+  def make_mine_job(amount)
+    MakeMineJob.perform_later(@width, @height, @board, amount)
+  end
+
+  def calculate_jobs
+    @x = (@number_of_mines / LIMIT).floor
+    z = LIMIT * @x
+    a = (@x+1)*LIMIT
+    b = a - @number_of_mines < LIMIT ? a - @number_of_mines : 0
+    @x_jobs = (LIMIT - b).to_i
+    @x_1_jobs = b.to_i
   end
```

Comment:
In the future this can be an algorithm that determines the optimal number of jobs to run given the number of buckets and the probability of random number collision for that bucket. As you increase the number of insertions you do in a single job, you increase the probability that one of those insertions will collide with an existing random number. When that happens the whole job is rolled back. At the moment its just at one since that works well for most requirements.